docs/src/routes/docs/components/button.svx

```diff
@@ -0,0 +1,127 @@
+    <Button outline disabled>button</Button>
+    <Button disabled>button</Button>
+
+    <Button filled danger>button</Button>
+    <Button outline danger>button</Button>
+    <Button danger noRipple>button</Button>
+
+    <Button filled>
+      <CheckIcon size="20" class="mr" />
+      icons
+    </Button>
+    <Button selected>selected</Button>
+    <Button round neutral><XIcon size="20" /></Button>
+  </div>
+  <div slot="source">
+
+```svelte
+<Button filled>button</Button>
+<Button outline>button</Button>
+<Button>button</Button>
+
+<Button filled disabled>button</Button>
+<Button outline disabled>button</Button>
+<Button disabled>button</Button>
+
+<Button filled danger>button</Button>
+<Button outline danger>button</Button>
+<Button danger noRipple>button</Button>
+
+<Button filled>
+  <CheckIcon size="20" class="mr" />
+  icons
+</Button>
+<Button selected>selected</Button>
+<Button round neutral><XIcon size="20" /></Button>
+```
+
+  </div>
+</Showcase>
+
+
+<H2 id="properties">Properties</H2>
+
+### Style Properties
+
+| <Label>Name</Label> | <Label>Type</Label> | <Label>Description</Label> |
+| --- | --- | --- |
+| **`filled`** | `boolean` | Adds a filled background to the button. <br /> Mutually exclusive with ==`outline`== and ==`selected`==. |
+| **`outline`** | `boolean` | Adds a border to the button. <br /> Mutually exclusive with ==`filled`==. |
+| **`danger`** | `boolean` | Gives the button a color of danger (red by default) <br /> Mutually exclusive with ==`neutral`==. |
+| **`neutral`** | `boolean` | Gives the button a neutral color (very dark gray by default). <br /> Mutually exclusive with ==`danger`==. |
+| **`round`** | `boolean` | Adjusts the padding to be equal on all sides. Great for icon-only buttons. |
+| **`rectangle`** | `boolean` | Halves the border radius and increases the padding. Well-suited for large menus. |
+| **`small`** | `boolean` | Cuts the padding to make the button smaller. |
+| **`selected`** | `boolean` | Gives the button a tinted background. Suitable for indicating the selected item. <br /> Mutually exclusive with ==`filled`==. |
+| **`noRipple`** | `boolean` | Disables the ripple effect. |
+
+### Functional Properties
+
+| <Label>Name</Label> | <Label>Type</Label> | <Label>Description</Label> |
+| --- | --- | --- |
+| **`disabled`** | `boolean` | Disables the button, styling it appropriately and disabling click events. <br /> Has no effect on link buttons (those with an ==`href`== prop). |
+| **`href`** | `string` | Turns the button into a link (prefetch-enabled for Sapper). |
+| **`noPrefetch`** | `string` | Disables Sapper prefetching for link buttons (those with an ==`href`== prop). |
+| **`events`** | `Array` | List of handlers for the [Events action](): `[{ name, handler }]`. |
+| **`...`** | `*` | Everything else will be passed to the underlying `<button>` or `<a>` element. |
+
+### Class Properties
+
+All `<Button>` components use the `.btn` class.
```

Comment:
Is this really something that should be documented? It's better to add their own classes if they want to override anything. Due to scoping, our selectors will have higher precedence over just `.btn {}` anyways, which can make it more frustrating when someone tries to modify it and can lead to lots of `!important`

Reply:
Coming back to our discussion about the importance of CSS. Yes, I whole-heartedly believe that this should be documented + the classes should have reasonable names (which they do, as of now). As for precedence, true, some might find it frustrating, but keep in mind that properly styling a child component in Svelte involves scoping the style from the outside:
```scss
.something-local :global .btn {}
```
which gets translated into 
```scss
.something-local.svelte-xxx .btn {}
```
which is both nicely isolated and has higher precedence over `.btn.svelte-yyy`

Reply:
"I believe" is not really an explanation/argument, but it's up to you)